base/src/androidLibMain/kotlin/ApplicationHolder.kt

```diff
@@ -25,7 +25,8 @@ class ApplicationHolder {
     companion object {
         var application: Application? = null
             set(application) {
-                checkNotNull(field) { "Application object can only be set once." }
+                if (field != null) 
```

Comment:
The logic should be inverted, isn't it?
So if we call set second time previous field value is not null.

`check(field != null) { "Application object can only be set once." }`

Reply:
contract for `check` is `Throws an IllegalStateException if the value is false.`
so `check (field == null)` (=**=**) is _true_ if it's the first time we set the field, so no exception as desired (and equivalent to what I have, but better syntax).

Reply:
Ah, yes, should be `==`

Reply:
I updated to this, but that's why this class should have tests (#90)